### PR TITLE
generate repository config name based on provider specific hash

### PR DESCRIPTION
**Pull request by HiteshRepo**
Enable user/client to be able to perform multiple backups in a concurrent manner irrespective of storage provider (s3/filesystem) for different tenants (anything that differentiates one account from another).

#### Does this PR need a docs update or release note?

- [ ] :white_check_mark: Yes, it's included
- [x] :clock1: Yes, but in a later PR
- [ ] :no_entry: No

#### Type of change

- [ ] :sunflower: Feature
- [x] :bug: Bugfix
- [ ] :world_map: Documentation
- [ ] :robot: Supportability/Tests
- [ ] :computer: CI/Deployment
- [ ] :broom: Tech Debt/Cleanup

#### Issue(s)

* #4443 

#### Test Plan

- [x] :muscle: Manual
- [x] :zap: Unit test
- [x] :green_heart: E2E

### CODE REVIEW

File: src/pkg/account/account.go
Comment (ryanfkeepers) [inline src/pkg/account/account.go:101]: Per error and logging conventions, lets keep templates out of the error message, please.  This can be handled as:
```suggestion
	return nil, clues.Stack(errInvalidProvider).With("provider", a.Provider)
```

File: src/internal/operations/maintenance_test.go
Comment (ryanfkeepers) [inline src/internal/operations/maintenance_test.go:43]: `uuid.NewString()[:7]` might be a good candidate to move into a package like `/str/testData` func.  That way all callers can call `strTD.Hash()`, and if we ever need to change the behavior, we only need to change it in the testData func instead of every place that generates a repo name hash.

File: src/pkg/repository/repository.go
Comment (ryanfkeepers) [inline src/pkg/repository/repository.go:168]:
```suggestion
		return clues.Wrap(err, "generating repo config")
```

File: src/pkg/repository/repository.go
Comment (ryanfkeepers) [inline src/pkg/repository/repository.go:369]: instead of having the account and storage return a config property set, why not have them just generate their own hash?  callers would have the option to concatenate hashes, or hash their union, both of which would still be sufficiently unique operations.  That would relieve a lot of the extra processing happening here (which also remove it as boilerplate if any other caller wants to use the funcs).

File: src/pkg/account/account.go
Comment (ryanfkeepers) [inline src/pkg/account/account.go:95]: can you please add a unit test that checks every account type to ensure it provides a hasher?  We want to make sure that on the next account type addition we'll automatically fail the test if it doesn't include a hashing func.

**Review by ryanfkeepers — APPROVED** [review]
Various minor things that need handling before merger, but good enough to approve and unblock.  Thanks for the work!

File: src/pkg/repository/repository.go
Comment (HiteshRepo) [reply src/pkg/repository/repository.go:369]: Yeah. Originally I has used 'md5' and not 'crc32'. So concatenating hashes would have made it big and trimming hash might not be a good idea as it might lead to collisions. But since I am using 'crc32' now, this should be a better choice. I will have this in the next commit.

**Pull request merged**
